sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -157,7 +157,10 @@ private static LiteralExpressionSyntax GetNullLiteralOrDefault(ArrowExpressionCl
 
         private static IEnumerable<ReturnStatementSyntax> GetReturnNullStatements(BlockSyntax methodBlock)
         {
-            return methodBlock.DescendantNodes(n => !n.IsKind(SyntaxKindEx.LocalFunctionStatement))
+            return methodBlock.DescendantNodes(n =>
```

Comment:
Ideally it should indeed be at a lower level, inside helpers/extensions. But I would not do this change inside this PR as it may have side effects on other rules.
I've updated the rule to take care of this though